### DISCUSSION TITLE
Login inside a github action

### DISCUSSION THREAD

**Discussion by javixeneize** [community/community#26344] [Actions]
<p>Hi! I am building a github action that uses a docker image. To download that image, it requires authentication.</p>
<p>I have tried to do a docker login in the wf before calling that action, but i am still receiving an unauthorised error. Any idea what should i do? Should i do the authentication inside the action itself?</p>
<p>My action has this inside:</p>
<blockquote>
<p>runs:<br>
using: ‘docker’<br>
image: ‘docker://myproject/miimage:latest’<br>
args:<br>
…</p>
</blockquote>
<p>How should i do the authentication?</p>
<p>Thanks</p>

**Comment by airtower-luna** [comment]
<p>I don’t think you can, the only options described in the documentation are using a Dockerfile in your repository, or using a <em>public</em> image:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-docker-actions" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-docker-actions" target="_blank" rel="noopener">Metadata syntax for GitHub Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by javixeneize** [comment]
<p>Thats a big limitation… is there any way to make it work with the pre tag? It is, do a docker login with a PAT sent as input? Would it work?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#pre-entrypoint"> <code>pre-entrypoint</code> </a> uses the same image, so I’m pretty sure it won’t. I also just noticed the <a href="https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/metadata-syntax-for-github-actions#runsimage"><code>runs.image</code></a> documentation explicitly says:</p>
<blockquote>
<p>The value can be the Docker base image name, a local <code>Dockerfile</code> in your repository, or a public image in Docker Hub or another registry.</p>
</blockquote>
<p>Which makes it pretty clear that private images are not supported.</p>

**Comment by javixeneize** [comment]
<p>Thanks. Do you know what is the best way to ask github if this will be supported? Do they usually answer here or do i need to raise an issue to any git where they manage new features?</p>

**Comment by airtower-luna** [comment]
<p>There are some Github people around here, so I hope they’ll see it. I don’t see any obvious category for feature requests on the <a href="https://support.github.com/request">contact page</a>.</p>
